GoogleDataTransport/GoogleDataTransport/Classes/GDTRegistrar.m

```diff
@@ -89,4 +89,39 @@ - (void)registerPrioritizer:(id<GDTPrioritizer>)prioritizer target:(GDTTarget)ta
   return targetToPrioritizer;
 }
 
+#pragma mark - GDTLifecycleProtocol
+
+- (void)appWillBackground:(nonnull UIApplication *)app {
+  dispatch_async(_registrarQueue, ^{
+    for (id<GDTUploader> uploader in [self->_targetToUploader allValues]) {
+      [uploader appWillBackground:app];
+    }
+    for (id<GDTPrioritizer> prioritizer in [self->_targetToPrioritizer allValues]) {
+      [prioritizer appWillBackground:app];
+    }
+  });
+}
+
+- (void)appWillForeground:(nonnull UIApplication *)app {
+  dispatch_async(_registrarQueue, ^{
+    for (id<GDTUploader> uploader in [self->_targetToUploader allValues]) {
+      [uploader appWillForeground:app];
+    }
+    for (id<GDTPrioritizer> prioritizer in [self->_targetToPrioritizer allValues]) {
+      [prioritizer appWillForeground:app];
+    }
+  });
+}
+
+- (void)appWillTerminate:(nonnull UIApplication *)app {
+  dispatch_sync(_registrarQueue, ^{
```

Comment:
This seems to make sense for this one to by sync and the others async, but want to confirm that it's intentional.

Reply:
It's intentional.